pygmt/src/grdcut.py

```diff
@@ -87,6 +87,18 @@ def grdcut(grid, **kwargs):
         - :class:`xarray.DataArray` if ``outgrid`` is not set
         - None if ``outgrid`` is set (grid output will be stored in file set by
           ``outgrid``)
+
+    Examples
+    --------
+    >>> import pygmt
+    >>> # Load a grid of @earth_relief_30m data, with an x-range of 10 to 30,
+    >>> # and a y-range of 15 to 25
+    >>> grid = pygmt.datasets.load_earth_relief(
+    ...     resolution="30m", region=[10, 30, 15, 25]
+    ... )
+    >>> # Create a new grid from an input grid, with an x-range of -3 to 1,
+    >>> # and a y-range of 2 to 5
+    >>> new_grid = pygmt.grdcut(grid=grid, region=[-3, 1, 2, 5])
```

Comment:
Thanks for moving forward with the in-line examples!

I would prefer that the example using a different dataset than the result from `load_earth_relief`. We could add a raster dataset to the proposed `load_sample_data` function (https://github.com/GenericMappingTools/pygmt/pull/1685). I think that `grid = load_sample_data(name="relief")` (hypothetically for `@tut_relief`) is more intuitive than relying on`load_earth_relief`.

Reply:
> I think that `grid = load_sample_data(name="relief")` (hypothetically for `@tut_relief.nc`) is more intuitive than relying on`load_earth_relief`.

Why do you think so? I think cutting a earth relief grid is a very common case, and users can simply use it without changes.

Reply:
I agree with @seisman on this. I think the examples are good to demonstrate how a user could get the code running, and it's better to demonstrate using the `load_earth_relief` for a usable example.